### PR TITLE
Drupal9-ready config directory

### PR DESCRIPTION
**Pull request by simesy**
See https://www.drupal.org/docs/8/configuration-management/changing-the-storage-location-of-the-sync-directory

This will update automatically with drupal 9, and both `drush cim` or `drush cim sync` will work interchangeably.

Any --partial import (eg from ../config/dev needs to use:

```
drush config:import -y --partial --source=../config/dev
```

### CODE REVIEW

**Comment by simesy** [comment]
@srowlands can i get a review on this please as I'd like to be doing this the right way on my current project.

**Comment by AlexSkrypnyk** [comment]
@simesy could you please resolve a conflict before this can be reviewed. thanks

**Comment by simesy** [comment]
If i can just confirm the name of the config directory then I have what I need and there is no urgency on the rest.

File: drupal/settings/all.settings.php [hunk 1]
```diff
@@ -17,8 +17,10 @@
 global $govcms_includes;
 $settings['container_yamls'][] = $govcms_includes . '/all.services.yml';
 
-// Config directory.
-$config_directories[CONFIG_SYNC_DIRECTORY] = '../config/sync';
+// Drupal 8 config.
+$config_directories[CONFIG_SYNC_DIRECTORY] = '../config/default';
+// Drupal 9 ready.
```
Comment (AlexSkrypnyk) [inline drupal/settings/all.settings.php]: this is actually Drupal >=8.8. Could you please update the PR accordingly.

File: drupal/settings/all.settings.php [hunk 2]
```diff
@@ -17,8 +17,10 @@
 global $govcms_includes;
 $settings['container_yamls'][] = $govcms_includes . '/all.services.yml';
 
-// Config directory.
-$config_directories[CONFIG_SYNC_DIRECTORY] = '../config/sync';
+// Drupal 8 config.
```
Comment (AlexSkrypnyk) [inline drupal/settings/all.settings.php]: Drupal <8.8

**Comment by simesy** [comment]
No at the moment govcms has two config directories `config/default` and `config/sync`. I just want to confirm which one is the default one going forward - since Drupal will only want one default one.

**Comment by simesy** [comment]
It's just a technically decision, someone senior needs to make it.

**Comment by AlexSkrypnyk** [comment]
@steveworley @srowlands keeping in mind that we have sites with existing configs in `config` dir, can change the location to `default` without causing issues?

**Comment by simesy** [comment]
@AlexSkrypnyk on all the saas sites I've worked with, exporting config goes into `config/default`.

**Comment by simesy** [comment]
So IMO `config/default` is the least risky, but the way the scaffold was ported, I notice the sync directory is getting priority now, so some new paas clients will potentially have their config in the sync directory, and so they would need to be advised.

**Comment by AlexSkrypnyk** [comment]
@simesy at least one of the PaaS clients have a `config` directory. We cannot just change this without providing a fallback. Maybe we need to introduce an env variable that allows to override this location for existing PaaS.

**Comment by simesy** [comment]
Ok just let me know what the correct directory will be.

**Comment by simesy** [comment]
I'm using config/default for now.

**Comment by thompsong** [comment]
We forked the govcms/scaffold-tooling into our paas projects and made these changes recently. 

https://github.com/govCMS/scaffold-tooling/blob/ea111da38688c0147cf847c856c1048af622a289/drupal/settings/all.settings.php#L20-L23

File: drupal/settings/all.settings.php [hunk 3]
```diff
@@ -17,8 +17,10 @@
...
+$config_directories[CONFIG_SYNC_DIRECTORY] = '../config/default';
```
Comment (AlexSkrypnyk) [inline drupal/settings/all.settings.php]: `$config_directories[CONFIG_SYNC_DIRECTORY] = getenv('CONFIG_SYNC_DIRECTORY') ?: '../config/default';`

File: drupal/settings/all.settings.php [hunk 4]
```diff
@@ -17,8 +17,10 @@
...
+// Drupal 9 ready.
+$settings['config_sync_directory'] = '../config/default';
```
Comment (AlexSkrypnyk) [inline drupal/settings/all.settings.php]: $settings[CONFIG_SYNC_DIRECTORY] = getenv('CONFIG_SYNC_DIRECTORY') ?: '../config/default';

**Comment by simesy** [comment]
`CONFIG_SYNC_DIRECTORY` is a deprecated drupal constant, so I wouldn't be using that. But in general I think that idea is ok.

**Comment by simesy** [comment]
@AlexSkrypnyk I guess what you're saying is: "it won't matter what the directory is, because it will be configurable."

So I'll use `config/default` like all the existing saas sites fwiw.

**Comment by simesy** [comment]
woop!

**Pull request merged**